src/func/collection.rs

```diff
@@ -22,8 +27,20 @@ pub struct Config {
 impl Config {
     /// Creates a new collection config with the given parameters.
     #[new]
-    pub fn new(ef_construction: usize, ef_search: usize, ml: f32) -> Self {
-        Self { ef_construction, ef_search, ml }
+    pub fn new(
+        ef_construction: usize,
+        ef_search: usize,
+        ml: f32,
+        distance: &str,
+    ) -> Result<Self, Error> {
+        Distance::from(&distance.into())?;
```

Comment:
I assume that this line is for validation?

Reply:
yes, we valid it when conf is created.